fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/util/Taggable2ObjectMap.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.content.registry.v1.util;
+
+import net.minecraft.tag.Tag;
+
+public interface Taggable2ObjectMap<K, V> {
```

Comment:
Isnt this just a map?

Reply:
It's also got a `add(Tag<K>, V)` method.

Reply:
Humm, fair point but I still think there are cleaner ways to handle `Taggable2ObjectMapRegistryImpl` then.

Reply:
it can still extend `Map` even if it has custom methods.